pysilcam/tracking/track.py

```diff
@@ -0,0 +1,417 @@
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Ws [cm/s]')
+
+    plt.sca(a[1, 1])
+    plt.plot(data['ECD [mm]'], data['Speed [cm/s]'], '.', color='0.8', alpha=0.1)
+    plt.xlabel('ECD [mm]')
+    plt.ylabel('Speed [cm/s]')
+
+    plt.show()
+
+
+def load_and_process(tracksfile, PIX_SIZE,
+                     minlength=0, maxlength=1e6, track_length_limit=15):
+    data = pd.read_hdf(tracksfile, 'Tracking/data')
+    tracks = post_process(data, PIX_SIZE,
+                          track_length_limit=track_length_limit,
+                          minlength=minlength, maxlength=maxlength)
+    return data, tracks
+
+
+def track_process(configfile, datapath, offset=0):
+    '''
+    This takes a silcam config file (and uses information within the [Tracking] section to track particles in images in
+    datapath. This will produce a -TRACKS.h5 file in the [General]datafile location (in a similar manner to normal
+    silcam processing).
+
+    Functions like pysilcam.tracking.track.make_boxplot can make plots with these -TRACK.h5 files.
+
+    :param configfile:
+    :param datapath:
+    :param offset:
+    :return:
+    '''
+    settings = PySilcamSettings(configfile)
+
+    sctr = Tracker()
+    sctr.av_window = settings.Background.num_images
+    sctr.MIN_LENGTH = settings.Tracking.min_length
+    sctr.MIN_SPEED = settings.Tracking.min_speed
+    sctr.GOOD_FIT = settings.Tracking.good_fit
+    sctr.THRESHOLD = settings.Process.threshold
+    sctr.ecd_tolerance = settings.Tracking.ecd_tolerance
+    sctr.PIX_SIZE = settings.PostProcess.pix_size
+
+    sctr.path = datapath
+    dataset_name = os.path.split(datapath)[-1] + '-TRACKS'
+    sctr.DATAFILE = os.path.join(settings.General.datafile, dataset_name)
+    sctr.track_length_limit = settings.Tracking.track_length_limit
+    sctr.initialise()
+    sctr.files = sctr.files[offset:]
+
+    os.makedirs(settings.General.datafile, exist_ok=True)
+
+    # setup HDF5 file and metadata
+    print('* Setting up', sctr.DATAFILE + '.h5')
+    with h5py.File(sctr.DATAFILE + '.h5', "a") as HDF5File:
+        meta = HDF5File.require_group('Meta')
+        meta.attrs['Modified'] = str(pd.datetime.now())
+        settings_dict = {s: dict(settings.config.items(s)) for s in settings.config.sections()}
+        meta.attrs['Settings'] = str(settings_dict)
+
+    sctr.process()
+
+
+def make_boxplot(tracksfile):
+    '''
+    given the input tracksfile (-TRACKs.h5), this function will look for all the other -TRACKS.h5 files in the same
+    folder and use this to create summary boxplots of tracked data, which is saved to the General.datafile location
+    specified within the settings contained in the last -TRACKS.h5 file.
+
+    :param tracksfile:
+    :return:
+    '''
+    h5filedir = os.path.split(tracksfile)[0]
+    h5file_list = glob.glob(os.path.join(h5filedir, '*-TRACKS.h5'))
+    dataset_names = [os.path.split(k)[-1].replace('-TRACKS.h5', '') for k in h5file_list]
```

Comment:
F405: 'os' may be undefined, or defined from star imports: pysilcam.tracking.silcamtracker  <!-- Automatically posted by Lintly -->